lib/adding_columns.py

```diff
@@ -91,13 +98,23 @@ def add_average_films_ratings(spark: SparkSession, data: DataFrame) -> DataFrame
         .select(["nconst", "average_films_rating"])
     )
     data = data.join(data_with_ratings, on="nconst", how="left")
+    data = data.na.fill(value = 0, subset = ["average_films_rating"])
     return data
 
+def add_normalized_columns(data: DataFrame) -> DataFrame:
+    unlist = udf(lambda x: round(float(list(x)[0]),3), DoubleType())
```

Comment:
poproszę o komentarz co ma robić ta lambda.

Reply:
dodane

Reply:
w jaki sposób vector zmienia się w double? Dasz mi jakiś przykład?

Reply:
funkcja VectorAssembler zamienia dane z kolumny na wektor, więc każda wartość w danej kolumnie jest wektorem, natomiast funkcja unlist będzie brała tę wartość typu wektor i zamieniała ją na double - bierze ona tylko pierwszą wartość w tym wektorze, ale ponieważ w funkcji VectorAssembler jako argument podaliśmy jedynie jedną kolumnę,  to w tych wektorach jest po jednej wartości.

Reply:
bardziej chodzi mi o przykład jak wyglądają wartości w kolumnie i jak są zamieniane.

Reply:
bigint (np. 1) -> vector ([1]) -> double (1)

Reply:
to po co konwersja na wektor jeśli tam jest jeden double?

Reply:
ponieważ MinMaxScaller działa na wektorach
![image](https://user-images.githubusercontent.com/116296531/212901494-842be8eb-be43-4504-a4b5-7ad68c22fedb.png)

Reply:
proszę dodaj odpowiedni komentarz w kodzie i możemy mergować. Możesz zrobić resolve tego wątku po dodaniu tego komentarza.-